sempahore test hang

TEST_CASE("semaphore 1 producers and many consumers", "[semaphore]")

```
BEGIN semaphore 1 producers and many consumers
1: producer 0 exiting
1: consumer 15 exiting
1: consumer 13 exiting
1: consumer 12 exiting
1: consumer 11 exiting
1: consumer 10 exiting
1: consumer 9 exiting
1: consumer 8 exiting
1: consumer 7 exiting
1: consumer 6 exiting
1: consumer 5 exiting
1: consumer 4 exiting
1: consumer 3 exiting
1: consumer 2 exiting
1: consumer 1 exiting
1: consumer 0 exiting
1/1 Test #1: libcoro_tests ....................***Timeout 1500.10 sec
0% tests passed, 1 tests failed out of 1
Total Test time (real) = 1500.11 sec
The following tests FAILED:
	  1 - libcoro_tests (Timeout)
Errors while running CTest
Output from these tests are in: /__w/libcoro/libcoro/build/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.
Error: Process completed with exit code 8.
```

--- a/include/coro/semaphore.hpp
+++ b/include/coro/semaphore.hpp
@@ -110,7 +110,7 @@ public:
                 {
                     return;
                 }
-            } while (m_counter.compare_exchange_weak(current, current + 1, std::memory_order::acq_rel, std::memory_order::acquire));
+            } while (!m_counter.compare_exchange_weak(current, current + 1, std::memory_order::acq_rel, std::memory_order::acquire));
         }
     }
 

--- a/test/test_semaphore.cpp
+++ b/test/test_semaphore.cpp
@@ -215,6 +215,7 @@ TEST_CASE("semaphore 1 producers and many consumers", "[semaphore]")
         [](std::shared_ptr<coro::thread_pool> tp, coro::semaphore<50>& s, std::atomic<uint64_t>& value, uint64_t id) -> coro::task<void>
     {
         co_await tp->schedule();
+        std::cerr << "consumer " << id << " starting\n";
 
         while (true)
         {

